File: fcc-python/binary_search.py
import math

def binary_search(list, target) :
    max = len(list) - 1
    min = 0

    while min < max :
        mid = math.floor(max + min / 2)
        if target == list[mid] :
            return mid
        elif target > mid :
            min = mid + 1
        else :
            max = mid - 1
    return None   

File: microsoft-typescript/src/binarySearch.ts
export function binarySearch (list: number[], target: number): number | null {
  let max: number = list.length - 1;
  let min: number = 0;

  while(min < max) {
    let mid: number = Math.floor(max + min / 2);

    if(list[mid] === target) {
      return mid;
    }
    else if (list[mid] < target) {
      min = mid + 1;
    }
    else {
      max = mid - 1;
    }
  }
  return null;
}
